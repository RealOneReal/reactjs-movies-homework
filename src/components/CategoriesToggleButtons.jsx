import { useState } from "react";
import { ToggleButtonGroup, ToggleButton } from "@mui/material";

const CategoriesToggleButtons = () => {
  const [categorie, setCategorie] = useState("popular");
  const handleChangeCategorie = (event) => {
    setCategorie(event.target.value);
  };

  const styles = {
    group: {
      float: "right",
      mt: 2,
      mr: 3,
    },
    button: {
      color: "secondary.light",
    },
  };

  return (
    <ToggleButtonGroup
      value={categorie}
      color="primary"
      onChange={handleChangeCategorie}
      exclusive
      sx={styles.group}
    >
      <ToggleButton value="popular" sx={styles.button}>
        Popular
      </ToggleButton>
      <ToggleButton value="top_rated" sx={styles.button}>
        Top rated
      </ToggleButton>
      <ToggleButton value="upcoming" sx={styles.button}>
        Upcoming
      </ToggleButton>
    </ToggleButtonGroup>
  );
};

export default CategoriesToggleButtons;
