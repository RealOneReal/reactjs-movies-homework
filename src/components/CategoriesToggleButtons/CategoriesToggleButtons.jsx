import { useDispatch, useSelector } from "react-redux";
import { useEffect } from "react";
import { ToggleButtonGroup, ToggleButton } from "@mui/material";
import {
  fetchGenresConfig,
  fetchMoviesByCategory,
} from "../../redux/movies/moviesThunks";
import { categoryChange } from "../../redux/search/searchSlice";
import { API_MOVIES } from "../../api/moviesAPI";

const CategoriesToggleButtons = () => {
  const dispatch = useDispatch();

  const { category } = useSelector((state) => state.search);
  const language = useSelector((state) => state.search.language);
  const inputValue = useSelector((state) => state.search.value);
  const handleChangeCategorie = ({ target }) => {
    if (inputValue) {
      return;
    }
    dispatch(categoryChange(target.value));
    dispatch(
      fetchMoviesByCategory({ category: target.value, pageNumber: 1, language })
    );
  };

  const styles = {
    group: {
      display: "flex",
      justifyContent: {
        md: "flex-end",
        xs: "center",
      },
      mr: 3,
    },
    button: {
      color: "secondary.light",
    },
  };

  useEffect(() => {
    if (inputValue) {
      return;
    }
    dispatch(fetchGenresConfig({ language }));
    dispatch(fetchMoviesByCategory({ category, pageNumber: 1, language }));
  }, [language]);

  return (
    <ToggleButtonGroup
      value={category}
      color="primary"
      onChange={handleChangeCategorie}
      exclusive
      sx={styles.group}
    >
      <ToggleButton value={API_MOVIES.POPULAR} sx={styles.button}>
        Popular
      </ToggleButton>
      <ToggleButton value={API_MOVIES.TOP_RATED} sx={styles.button}>
        Top rated
      </ToggleButton>
      <ToggleButton value={API_MOVIES.UPCOMING} sx={styles.button}>
        Upcoming
      </ToggleButton>
    </ToggleButtonGroup>
  );
};

export default CategoriesToggleButtons;
