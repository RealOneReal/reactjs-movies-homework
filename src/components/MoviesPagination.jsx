import { Pagination, PaginationItem } from "@mui/material";

const MoviesPagination = (props) => {
  // todo: realize pagination after connecting Redux
  const { total_pages, page } = props;
  return (
    <Pagination
      count={total_pages}
      page={page}
      hideNextButton={true}
      hidePrevButton={true}
      color="primary"
      renderItem={(item) => (
        <PaginationItem
          component="button"
          sx={{ color: "primary.dark" }}
          {...item}
        />
      )}
    />
  );
};

export default MoviesPagination;
