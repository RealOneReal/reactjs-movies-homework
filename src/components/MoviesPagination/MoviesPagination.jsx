import { Pagination, PaginationItem } from "@mui/material";

const MoviesPagination = ({ total_pages, page }) => {
  // todo: realize pagination after connecting Redux
  return (
    <Pagination
      count={total_pages}
      page={page}
      hideNextButton={true}
      hidePrevButton={true}
      color="primary"
      renderItem={(item) => (
        <PaginationItem
          component="button"
          sx={{ color: "primary.dark" }}
          {...item}
        />
      )}
    />
  );
};

export default MoviesPagination;
