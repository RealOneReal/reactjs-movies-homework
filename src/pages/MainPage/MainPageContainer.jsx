import MovieCardList from "../../components/MovieCardList";
import DataFromTMDB from "../../mock-data/mockMovies";
const MainPageContainer = () => {
  // todo: add mockData from MovieDB
  return <MovieCardList {...DataFromTMDB} />;
};
export default MainPageContainer;
