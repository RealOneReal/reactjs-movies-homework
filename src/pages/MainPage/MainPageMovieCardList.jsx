import { Grid } from "@mui/material";
import MovieCard from "../../components/MovieCard/MovieCard";
import DataFromTMDB from "../../mock-data/mockMovies";

const MainPageMovieCardList = () => {
  // todo: add mockData from MovieDB
  const results = DataFromTMDB.results;

  return (
    <Grid container spacing={1} justifyContent="center">
      {results &&
        results.map((card) => (
          <Grid item key={card.id}>
            <MovieCard {...card} />
          </Grid>
        ))}
    </Grid>
  );
};
export default MainPageMovieCardList;
