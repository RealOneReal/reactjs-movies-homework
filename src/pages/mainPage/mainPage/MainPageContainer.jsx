import { Grid } from "@mui/material";

const MainPageContainer = () => {};
