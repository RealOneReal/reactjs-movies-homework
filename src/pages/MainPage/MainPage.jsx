import { Box, Container } from "@mui/material";
import MainPageMovieCardList from "./MainPageMovieCardList";
import MoviesPagination from "../../components/MoviesPagination/MoviesPagination";
import CategoriesToggleButtons from "../../components/CategoriesToggleButtons/CategoriesToggleButtons";

const MainPage = () => {
  const { totalPages } = useSelector((state) => state.movies);
  const { pageNumber } = useSelector((state) => state.search);
  return (
    <Box sx={{ backgroundColor: "secondary.dark" }}>
      <Container maxWidth="lg" sx={{ pt: 2 }}>
        <CategoriesToggleButtons />
        <MainPageMovieCardList />
        <Box
          sx={{
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
            padding: " 16px 0",
          }}
        >
          <MoviesPagination pageNumber={pageNumber} totalPages={totalPages} />
        </Box>
      </Container>
    </Box>
  );
};
export default MainPage;
