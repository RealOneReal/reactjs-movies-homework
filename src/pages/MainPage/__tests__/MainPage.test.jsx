import { render } from "@testing-library/react";
import MainPage from "../MainPage";
import MainPageContainer from "../MainPageContainer";
import CategoriesToggleButtons from "../../../components/CategoriesToggleButtons/CategoriesToggleButtons";
import MoviesPagination from "../../../components/MoviesPagination/MoviesPagination";

jest.mock("../MainPageContainer", () => () => "MainPageContainer");
jest.mock(
  "../../../components/CategoriesToggleButtons/CategoriesToggleButtons",
  () => () => "CategoriesToggleButtons"
);
jest.mock(
  "../../../components/MoviesPagination/MoviesPagination",
  () => () => "MoviesPaggination"
);

describe("MainPage", () => {
  it("should be rendered", () => {
    const { container } = render(<MainPage />);
    expect(container).toMatchSnapshot();
  });
});
