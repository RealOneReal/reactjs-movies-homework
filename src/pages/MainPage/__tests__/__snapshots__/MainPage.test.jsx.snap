// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage should be rendered 1`] = `
<div>
  <div
    class="MuiBox-root css-4ozt90"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-r5r38p-MuiContainer-root"
    >
      CategoriesToggleButtons
      MainPageContainer
      <div
        class="MuiBox-root css-2hoarh"
      >
        MoviesPaggination
      </div>
    </div>
  </div>
</div>
`;
